test/test_scheduled.rb

```diff
@@ -79,6 +79,30 @@ def call(worker_class, job, queue, r)
       end
     end
 
+    it 'should not enqueue jobs when terminate has been called' do
```

Comment:
I just added this test for the terminate case because I was struggling to find a clean way to write other tests that stub and test the behavior inside the `enqueue_jobs` loop.

I think that would be testing implementation details anyway so maybe this is fine?